tests/lib/rules/no-dupe-disjunctions.ts

```diff
@@ -391,6 +391,86 @@ tester.run("no-dupe-disjunctions", rule as any, {
                 "Unexpected useless alternative. This alternative is a strict subset of '\\d|[a-z]|_' and can be removed.",
             ],
         },
+        {
+            code: String(/a+|a|b|c/),
+            errors: [
+                {
+                    message:
+                        "Unexpected useless alternative. This alternative is a strict subset of 'a+' and can be removed.",
+                    column: 5,
+                },
+            ],
+        },
+        {
+            code: String(/((?:ab|ba)|(?:ba|ac))/),
+            errors: [
+                {
+                    message:
+                        "Unexpected useless element. All paths of '(?:ba|ac)' that go through this element are a strict subset of '(?:ab|ba)'. This element can be removed.",
+                    column: 16,
+                },
+            ],
+        },
+        {
+            code: String(/a+|(?:a|b|c)/),
+            errors: [
+                {
+                    message:
+                        "Unexpected useless element. All paths of '(?:a|b|c)' that go through this element are a strict subset of 'a+'. This element can be removed.",
+                    column: 8,
+                },
+            ],
+        },
+        {
+            code: String(/a+|[abc]/),
+            errors: [
+                {
+                    message:
+                        "Unexpected useless element. All paths of '[abc]' that go through this element are a strict subset of 'a+'. This element can be removed.",
+                    column: 6,
+                },
+            ],
+        },
+        {
+            code: String(/a+|[a-c]/),
+            errors: [
+                {
+                    message:
+                        "Unexpected useless element. All paths of '[a-c]' that go through this element are a strict subset of 'a+'. This element can be removed.",
+                    column: 6,
+                },
+            ],
+        },
+        {
+            code: String(/a|aa|ba/),
+            errors: [
+                {
+                    message:
+                        "Unexpected useless alternative. This alternative is already covered by 'a' and can be removed.",
+                    column: 4,
+                },
+            ],
+        },
+        {
+            code: String(/a|(a|b)a/),
+            errors: [
+                {
+                    message:
+                        "Unexpected useless element. All paths of '(a|b)a' that go through this element are already covered by 'a'. This element can be removed. Careful! This alternative contains capturing groups which might be difficult to remove.",
```

Comment:
The capturing group warning is incorrect. I'll fix it.

Reply:
Again, sorry for the inconvenience. The warning has been fixed in b42349108f6d7c099413d054eb4d88fa7eea39ec.